src/Control/Carrier/Error/Either.hs

```diff
@@ -11,62 +11,37 @@ module Control.Carrier.Error.Either
 , run
 ) where
 
-import Control.Applicative (Alternative(..), liftA2)
+import Control.Applicative (Alternative(..))
 import Control.Carrier
 import Control.Effect.Error
 import Control.Monad (MonadPlus(..), (<=<))
 import qualified Control.Monad.Fail as Fail
 import Control.Monad.Fix
 import Control.Monad.IO.Class
 import Control.Monad.Trans.Class
+import Control.Monad.Trans.Except
 
 -- | Run an 'Error' effect, returning uncaught errors in 'Left' and successful computations’ values in 'Right'.
 --
 --   prop> run (runError (pure a)) === Right @Int @Int a
 runError :: ErrorC exc m a -> m (Either exc a)
-runError = runErrorC
+runError = runExceptT . runErrorC
 
-newtype ErrorC e m a = ErrorC { runErrorC :: m (Either e a) }
-  deriving (Functor)
+newtype ErrorC e m a = ErrorC { runErrorC :: ExceptT e m a }
+  deriving (Applicative, Functor, Monad, Fail.MonadFail, MonadFix, MonadIO, MonadTrans)
```

Comment:
Defining `ErrorC` atop `ExceptT` simultaneously fixes the unlawful `Applicative` instance and allows us to remove a great number of hand-written instances.

Reply:
It also fixes the divergent `MonadFix` instance.